BaseApp/PAG99999.txt

```diff
@@ -0,0 +1,108 @@
+OBJECT Page 99999 Camera
+{
+  OBJECT-PROPERTIES
+  {
+    Date=17-10-20;
+    Time=16:17:10;
+    Modified=Yes;
+    Version List=;
+  }
+  PROPERTIES
+  {
+    Editable=No;
+    CaptionML=ENU=Camera;
+    SourceTable=Table99998;
+    PageType=StandardDialog;
+    SourceTableTemporary=Yes;
+    OnOpenPage=BEGIN
+                 CameraAvailable := CameraProvider.IsAvailable();
+
+                 IF CameraAvailable THEN BEGIN
+                   CameraOptions := CameraOptions.CameraOptions;
+                   CameraOptions.Quality := Quality;
+                   CameraOptions.AllowEdit := AllowEdit;
+
+                   CASE EncodingType OF
+                     EncodingType::JPEG:
+                       CameraOptions.EncodingType := 'JPEG';
+                     EncodingType::PNG:
+                       CameraOptions.EncodingType := 'PNG';
+                   END;
+
+                   CASE MediaType OF
+                     MediaType::AllMedia:
+                       CameraOptions.MediaType := 'AllMedia';
+                     MediaType::Picture:
+                       CameraOptions.MediaType := 'Picture';
+                     MediaType::Video:
+                       CameraOptions.MediaType := 'Video';
+                   END;
+
+                   CASE SourceType OF
+                     SourceType::PhotoLibrary:
+                       CameraOptions.SourceType := 'PhotoLibrary';
+                     SourceType::Camera:
+                       CameraOptions.SourceType := 'Camera';
+                   END;
+
+                   CameraProvider := CameraProvider.Create();
+                   CameraProvider.RequestPictureAsync(CameraOptions);
+                 END;
+               END;
+
```

Comment:
Us QueryClose to return the LookUpOK if you got a picture.

Reply:
Not possible, as discussed. Instead added GetPicture(stream: InStream): boolean, as agreed.